app/src/main/java/ch/epfl/sweng/SDP/auth/LoginActivity.java

```diff
@@ -74,8 +75,15 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                 assert response != null;
                 handleSuccessfulSignIn(response);
             } else {
+                // User pressed the back button
+                if (response == null) {
+                    launchActivity(MainActivity.class);
```

Comment:
If you go back to the MainActivity you should add overridePendingTransition(R.anim.fade_in, R.anim.fade_out); in MainActivity after super.onCreate(...); for a smoother transition but that's not very important I can do it later in another PR

Reply:
Alright.